### PR TITLE
Add sonata events to be able to include javascript with events

### PR DESCRIPTION
**Pull request by maximehuran**
I want to be able to create a plugin compatible with Sylius and Monofony.
In order to to that, it could be nice to add these events to fit with Sylius.

Admin : https://docs.sylius.com/en/1.6/cookbook/frontend/admin-js-and-css.html
```yaml
# config/services.yaml
services:
    app.block_event_listener.admin.layout.javascripts:
        class: Sylius\Bundle\UiBundle\Block\BlockEventListener
        arguments:
            - '@@App/Admin/_javascripts.html.twig'
        tags:
            - { name: kernel.event_listener, event: sonata.block.event.sylius.admin.layout.javascripts, method: onBlockEvent }
```

In front : 
```yaml
# config/services.yaml
services:
    app.block_event_listener.shop.layout.javascripts:
        class: Sylius\Bundle\UiBundle\Block\BlockEventListener
        arguments:
            - '@@App/Shop/_javascripts.html.twig'
        tags:
            - { name: kernel.event_listener, event: sonata.block.event.sylius.shop.layout.javascripts, method: onBlockEvent }
```

### CODE REVIEW

**Comment by loic425** [comment]
@maximehuran Maybe can you it for stylesheets too?

**Comment by loic425** [comment]
Thank you @maximehuran! 🎉

**Comment by maximehuran** [comment]
I will do it in another PR ;)

File: templates/frontend/layout.html.twig [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
 {% block javascripts %}
     <script type="text/javascript" src="{{ asset('assets/frontend/app.js', 'frontend') }}"></script>
+    {{ sonata_block_render_event('sylius.shop.layout.javascripts') }}
```
Comment (loevgaard) [inline templates/frontend/layout.html.twig]: What is `shop` in the realm of Monofony?

Comment (maximehuran) [reply templates/frontend/layout.html.twig]: We could change it but with that we can call the same event for a Sylius Plugin and a Monofony one

Comment (loic425) [reply templates/frontend/layout.html.twig]: Yes "shop" term is replaced with "app" on Monofony (AppUser for example, ) but to fit with sylius events for plugin, it's coherent.

Comment (maximehuran) [reply templates/frontend/layout.html.twig]: Maybe we can rename it with `app` and a plugin which need to be compatible with Sylius and Monofony will listen `sylius.shop.layout.javascripts` and `sylius.app.layout.javascripts` ?

Comment (loevgaard) [reply templates/frontend/layout.html.twig]: That certainly would be better imo.

Comment (loic425) [reply templates/frontend/layout.html.twig]: maybe frontend and backend to fit with assets?

**Pull request merged**